test/qpy_compat/process_version.sh

```diff
@@ -29,6 +29,11 @@ if [[ ${parts[0]} -eq 0 && ${parts[1]} -lt 18 ]] ; then
     exit 0
 fi
 
+# Exclude any non-rc pre-releases as they don't have stable API guarantees
+if [[ $version == *"b"* || $version == *"a"* ]] ; then
+    exit 0
+fi
+
 # If the source version is newer than the version under test exit fast because
 # there is no QPY compatibility for loading a qpy file generated from a newer
 # release with an older release of Qiskit.
```

Comment:
Just below these lines we call out to a Python script `compare_versions.py`, which makes a general comparison.  Maybe we ought to move the logic to there, and handle proper version normalisation in Python with `packaging.version`?  We _intend_ to only use the canonical forms of versions, but the logic's all going to be easier to verify if it's in Python space with typed objects.

Reply:
I can move it to python space, but `compare_versions.py` isn't actually using `packaging.version` (to avoid needing to check the dependency) it's just using the regex to match the version parts. So I'd probably just basically be doing `'a' in match.group('pre') or 'b' in  match.group('pre')`

Reply:
Oh, fair enough, in that case let's just leave it in bash.